Traceback (most recent call last):
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
ohe_columns = list(
    preprocessor.named_transformers_["pipeline-4"]
    .named_steps["onehotencoder"]
    .get_feature_names(categorical_features)
)
new_columns = (
    numeric_features + ordinal_features_reg + ordinal_features_oth + ohe_columns
)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
Cell [0;32mIn[15], line 4[0m
[1;32m      1[0m ohe_columns [38;5;241m=[39m [38;5;28mlist[39m(
[1;32m      2[0m     [43mpreprocessor[49m[38;5;241;43m.[39;49m[43mnamed_transformers_[49m[43m[[49m[38;5;124;43m"[39;49m[38;5;124;43mpipeline-4[39;49m[38;5;124;43m"[39;49m[43m][49m
[1;32m      3[0m [43m    [49m[38;5;241;43m.[39;49m[43mnamed_steps[49m[43m[[49m[38;5;124;43m"[39;49m[38;5;124;43monehotencoder[39;49m[38;5;124;43m"[39;49m[43m][49m
[0;32m----> 4[0m [43m    [49m[38;5;241;43m.[39;49m[43mget_feature_names[49m(categorical_features)
[1;32m      5[0m )
[1;32m      6[0m new_columns [38;5;241m=[39m (
[1;32m      7[0m     numeric_features [38;5;241m+[39m ordinal_features_reg [38;5;241m+[39m ordinal_features_oth [38;5;241m+[39m ohe_columns
[1;32m      8[0m )

[0;31mAttributeError[0m: 'OneHotEncoder' object has no attribute 'get_feature_names'

